## Kaggle Exercise

The data used for this test is sourced from the Museum of Modern Art data on Kaggle: https://www.kaggle.com/momanyc/museum-collection



If you are comfortable in more than one, please feel free to share one or more solutions in a different language.
1. 
    Which artist in this data set lived the lo
    ngest2. ?
    Who are the top 10 artists by the number of ar
    twork3. s?
    Which artist is created the most artwork by total surfa
    ce ar4. ea?
    Did any artists have artwork acquired during their lew cluster.


## Libraries and loading the dataset 

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np
import re
# Warnings
import warnings
warnings.filterwarnings("ignore")

In [2]:
# Cargo dataset
dfa = pd.read_csv("artists.csv")
dfb = pd.read_csv("artworks.csv")

## 0. Getting to know our data

### dfa = Artist's Dataframe

In [3]:
dfa.head(2)

Artist ID            Name Nationality Gender  Birth Year  Death Year
0          1  Robert Arneson    American   Male      1930.0      1992.0
1          2  Doroteo Arnaiz     Spanish   Male      1936.0         NaN

In [4]:
dfa.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15091 entries, 0 to 15090
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Artist ID    15091 non-null  int64  
 1   Name         15091 non-null  object 
 2   Nationality  12603 non-null  object 
 3   Gender       12019 non-null  object 
 4   Birth Year   11237 non-null  float64
 5   Death Year   4579 non-null   float64
dtypes: float64(2), int64(1), object(3)
memory usage: 707.5+ KB


In [5]:
dfa.shape

(15091, 6)

In [6]:
dfa.isnull().sum()

Artist ID          0
Name               0
Nationality     2488
Gender          3072
Birth Year      3854
Death Year     10512
dtype: int64

### dfb = Artwork's Dataframe

In [7]:
dfb.head(2)

Artwork ID                                              Title Artist ID  \
0           2  Ferdinandsbrücke Project, Vienna, Austria, Ele...      6210   
1           3  City of Music, National Superior Conservatory ...      7470   

                       Name  Date  \
0               Otto Wagner  1896   
1  Christian de Portzamparc  1987   

                                          Medium  \
0  Ink and cut-and-pasted painted pages on paper   
1              Paint and colored pencil on print   

                           Dimensions Acquisition Date  \
0  19 1/8 x 66 1/2" (48.6 x 168.9 cm)       1996-04-09   
1       16 x 11 3/4" (40.6 x 29.8 cm)       1995-01-17   

                                              Credit Catalogue  ...  \
0  Fractional and promised gift of Jo Carole and ...         Y  ...   
1  Gift of the architect in honor of Lily Auchinc...         Y  ...   

  Classification Object Number Diameter (cm)  Circumference (cm)  Height (cm)  \
0   Architecture      885.1996           NaN                 NaN      48.6000   
1   Architecture        1.1995           NaN                 NaN      40.6401   

   Length (cm)  Width (cm)  Depth (cm)  Weight (kg)  Duration (s)  
0          NaN    168.9000         NaN          NaN           NaN  
1          NaN     29.8451         NaN          NaN           NaN  

[2 rows x 21 columns]

In [8]:
dfb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 130262 entries, 0 to 130261
Data columns (total 21 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   Artwork ID          130262 non-null  int64  
 1   Title               130210 non-null  object 
 2   Artist ID           128802 non-null  object 
 3   Name                128802 non-null  object 
 4   Date                127950 non-null  object 
 5   Medium              118343 non-null  object 
 6   Dimensions          118799 non-null  object 
 7   Acquisition Date    124799 non-null  object 
 8   Credit              127192 non-null  object 
 9   Catalogue           130262 non-null  object 
 10  Department          130262 non-null  object 
 11  Classification      130262 non-null  object 
 12  Object Number       130262 non-null  object 
 13  Diameter (cm)       1399 non-null    float64
 14  Circumference (cm)  10 non-null      float64
 15  Height (cm)         111893 non-nul

In [9]:
dfb.shape

(130262, 21)

In [10]:
dfb.isnull().sum()

Artwork ID                 0
Title                     52
Artist ID               1460
Name                    1460
Date                    2312
Medium                 11919
Dimensions             11463
Acquisition Date        5463
Credit                  3070
Catalogue                  0
Department                 0
Classification             0
Object Number              0
Diameter (cm)         128863
Circumference (cm)    130252
Height (cm)            18369
Length (cm)           129526
Width (cm)             19259
Depth (cm)            118819
Weight (kg)           129964
Duration (s)          127178
dtype: int64

In [11]:
(130262-128863)/130262*100

1.0739893445517494

In [12]:
dfb.columns

Index(['Artwork ID', 'Title', 'Artist ID', 'Name', 'Date', 'Medium',
       'Dimensions', 'Acquisition Date', 'Credit', 'Catalogue', 'Department',
       'Classification', 'Object Number', 'Diameter (cm)',
       'Circumference (cm)', 'Height (cm)', 'Length (cm)', 'Width (cm)',
       'Depth (cm)', 'Weight (kg)', 'Duration (s)'],
      dtype='object')

## 1. Which artist in this data set lived the longest?

In [13]:
# Create a copy of the Dataframe for each exercise
dfa1 = dfa.copy()

In [14]:
# The question is made in past tense therefore the artists that we are being asked for are not alive anymore so we will delete most of the artists whose Death Year is null
# As we need to know when where they born in order to determine the age, we will also delete those whose Birth Year is null
dfa1.dropna(subset=['Birth Year', 'Death Year'], inplace=True)

In [15]:
#Last lived year by the most recent artist
dfa1["Death Year"].max()

2017.0

In [16]:
#Substract the Birth Year from Death Year in order to get how many years the artist lived
dfa1["Lived"] = dfa1["Death Year"] - dfa1["Birth Year"]

In [17]:
dfa1

Artist ID                 Name Nationality Gender  Birth Year  \
0              1       Robert Arneson    American   Male      1930.0   
9             11      Jean (Hans) Arp      French   Male      1886.0   
11            13  J. Arrelano Fischer     Mexican   Male      1911.0   
12            15        Folke Arstrom     Swedish   Male      1907.0   
13            16    Cristobal Arteche     Spanish   Male      1900.0   
...          ...                  ...         ...    ...         ...   
15035      67127             John Sex    American    NaN      1956.0   
15036      67128      Egidio Bonfante     Italian   Male      1922.0   
15072      67336           Cal Dalton    American    NaN      1908.0   
15073      67337           Cal Howard    American    NaN      1911.0   
15079      67350    Howard Guttenplan    American    NaN      1934.0   

       Death Year  Lived  
0          1992.0   62.0  
9          1966.0   80.0  
11         1995.0   84.0  
12         1997.0   90.0  
13         1964.0   64.0  
...           ...    ...  
15035      1990.0   34.0  
15036      2004.0   82.0  
15072      1974.0   66.0  
15073      1993.0   82.0  
15079      2015.0   81.0  

[4566 rows x 7 columns]

In [18]:
# Order by lived years, in a descending order the df.
dfa1.sort_values(by='Lived', ascending=False)

Artist ID                                               Name  \
12486      38294  Union Paper Bag Machine Company, Philadelphia, PA   
3449        3844                                      Leni Matthaei   
9164       23931                                          B. Efimov   
13568      43532                                 Manoel de Oliveira   
9306       24431                                    Horacio Coppola   
...          ...                                                ...   
14570      48051                                     Memphis, Milan   
14245      47019                                       Joy Division   
14918      49496                                  Joshua Light Show   
12496      38423                            Save The Press Campaign   
12109      36946                                      Hi Red Center   

      Nationality  Gender  Birth Year  Death Year  Lived  
12486         NaN     NaN      1869.0      1999.0  130.0  
3449       German  Female      1873.0      1981.0  108.0  
9164      Russian    Male      1900.0      2008.0  108.0  
13568  Portuguese     NaN      1908.0      2015.0  107.0  
9306    Argentine    Male      1906.0      2012.0  106.0  
...           ...     ...         ...         ...    ...  
14570         NaN     NaN      1981.0      1985.0    4.0  
14245     British     NaN      1976.0      1980.0    4.0  
14918    American     NaN      1967.0      1970.0    3.0  
12496         NaN     NaN      1988.0      1990.0    2.0  
12109         NaN     NaN      1963.0      1964.0    1.0  

[4566 rows x 7 columns]

__________

With this line of code we see that the results in ascending order leads us to:

`1st position` Union Paper Bag Machine Company having lived **130 years** 

`2nd position`Leni Matthaei with **108 years lived***

_*After a quick search on Google we confirm that Leni Matthaei lived 108 years_

________

##     2. Who are the top 10 artists by the number of artworks?


In [19]:
dfb2 = dfb.copy()

In [20]:
# We saw that all the artworks have an artist name so we have no nulls on names therefore we can proceed to
# Create a list with the artists names
artists_list = list(dfb2.Name.unique())

In [21]:
# We have 13531 different artists (we have ~ 130K artworks)
len(artists_list)

13531

In [22]:
# This will give us the response to the question
dfb2['Name'].value_counts().head(10)

Eugène Atget                5050
Louise Bourgeois            3318
Ludwig Mies van der Rohe    2566
Unknown photographer        1575
Jean Dubuffet               1435
Lee Friedlander             1317
Pablo Picasso               1310
Marc Chagall                1162
Henri Matisse               1063
Pierre Bonnard               894
Name: Name, dtype: int64

##    3. Which artist is created the most artwork by total surface area?


In [23]:
dfb3 = dfb.copy()

In [24]:
dfb3.head(1)

Artwork ID                                              Title Artist ID  \
0           2  Ferdinandsbrücke Project, Vienna, Austria, Ele...      6210   

          Name  Date                                         Medium  \
0  Otto Wagner  1896  Ink and cut-and-pasted painted pages on paper   

                           Dimensions Acquisition Date  \
0  19 1/8 x 66 1/2" (48.6 x 168.9 cm)       1996-04-09   

                                              Credit Catalogue  ...  \
0  Fractional and promised gift of Jo Carole and ...         Y  ...   

  Classification Object Number Diameter (cm)  Circumference (cm)  Height (cm)  \
0   Architecture      885.1996           NaN                 NaN         48.6   

   Length (cm)  Width (cm)  Depth (cm)  Weight (kg)  Duration (s)  
0          NaN       168.9         NaN          NaN           NaN  

[1 rows x 21 columns]

In [25]:
# Drop null row where we do not have the artist name as we need it to answer our question
dfb3.dropna(subset=['Name'], inplace=True)

Taking into account that 14% of thte data has the Height value missing, we will only operate with the ones that we have the measurements

In case we wouldn't have the Height neither the Width columns, we would have to extract that information from the Dimensions column as it follows:
* Get the information we have in our column 'Dimensions' inside the ( ) and once we have it, we will have to split that by the numers separated by the 'x'.
* Later we calculate the surface of the artwork and then we will add all the surfaces of the same artist and get the one with the biggest surface area.

As we have different types of artworks, they are calculated differently, so we will calculate the surface with the data provided by the dataset as it follows:

* Rectangular forms (paints mostly): Height * Width
* 

In [26]:
# We have the 98.92% of the artworks with Nans or 0 values in Diameter or Circumference, therefore we will calculate the 
dfb3[(dfb3['Diameter (cm)'].isnull() | dfb3['Diameter (cm)'].eq(0)) & (dfb3['Circumference (cm)'].isnull() | dfb3['Circumference (cm)'].eq(0))].shape

(127405, 21)

In [27]:
128859 /dfb3.shape[0]

1.0004425397121162

In [28]:
#Function that multiplies height and width if they are not null
def surface_calc(row):
    if not pd.isnull(row['Height (cm)']) and not pd.isnull(row['Width (cm)']):
        return row['Height (cm)'] * row['Width (cm)']
    elif not pd.isnull(row['Height (cm)']) and not pd.isnull(row['Width (cm)']) and not pd.isnull(row['Depth (cm)']):    
        return 2 * (row['Height (cm)'] * row['Width (cm)'] + row['Width (cm)'] * row['Depth (cm)'] + row['Height (cm)'] * row['Depth (cm)'])
    elif not pd.isnull(row['Diameter (cm)']):
        diameter = row['Diameter (cm)']
        radius = diameter / 2
        surface_area = 4 * np.pi * (radius ** 2)
        return surface_area
    elif not pd.isnull(row['Circumference (cm)']):
        circumference = row['Circumference (cm)']
        radius = circumference / (2 * np.pi)
        surface_area = np.pi * (radius ** 2)
        return surface_area
    else:
        return np.nan
# Creat a column where we apply the function
dfb3['Surface (cm^2)'] = dfb3.apply(surface_calc, axis=1).round(2) #  apply pasa automáticamente cada fila del DataFrame a la función surface_calc como parámetro. 
dfb3['Surface (m^2)'] = (dfb3['Surface (cm^2)']/10000).round(2)

In [29]:
# We have the artwork with the biggest surface
dfb3.sort_values(by='Surface (m^2)', ascending=False).head(1)

Artwork ID     Title Artist ID             Name       Date  \
33718       35639  Untitled       902  James Lee Byars  (1962-64)   

                                                  Medium  \
33718  Japanese handmade white flax paper, hinged and...   

                                              Dimensions Acquisition Date  \
33718  Folded: 36 x 36 x 3" (91 x 91 x 7.6 cm) Unfold...       1966-06-07   

                   Credit Catalogue  ... Diameter (cm) Circumference (cm)  \
33718  Gift of the artist         Y  ...         274.3                NaN   

      Height (cm)  Length (cm)  Width (cm)  Depth (cm)  Weight (kg)  \
33718      1097.3          NaN      9144.0         NaN          NaN   

       Duration (s)  Surface (cm^2)  Surface (m^2)  
33718           NaN      10033711.2        1003.37  

[1 rows x 23 columns]

##    4. Did any artists have artwork acquired during their lifetime?


In [25]:
dfb4 = dfb.copy()
dfa4 = dfa.copy()

In [26]:
dfa4

Artist ID             Name Nationality Gender  Birth Year  Death Year
0              1   Robert Arneson    American   Male      1930.0      1992.0
1              2   Doroteo Arnaiz     Spanish   Male      1936.0         NaN
2              3      Bill Arnold    American   Male      1941.0         NaN
3              4  Charles Arnoldi    American   Male      1946.0         NaN
4              5      Per Arnoldi      Danish   Male      1941.0         NaN
...          ...              ...         ...    ...         ...         ...
15086      67452      Liu Jianhua     Chinese   Male      1962.0         NaN
15087      67453         Leng Lin     Chinese   Male      1965.0         NaN
15088      67652     Ellie Nagler         NaN    NaN         NaN         NaN
15089      67694   Glenn Williams         NaN   Male         NaN         NaN
15090      67695        Ely Ramen         NaN    NaN         NaN         NaN

[15091 rows x 6 columns]

In [27]:
dfb4.columns

Index(['Artwork ID', 'Title', 'Artist ID', 'Name', 'Date', 'Medium',
       'Dimensions', 'Acquisition Date', 'Credit', 'Catalogue', 'Department',
       'Classification', 'Object Number', 'Diameter (cm)',
       'Circumference (cm)', 'Height (cm)', 'Length (cm)', 'Width (cm)',
       'Depth (cm)', 'Weight (kg)', 'Duration (s)'],
      dtype='object')

In [28]:
# Drop the columns we do not need for answering this question

dfb4.drop(columns=['Credit', 'Catalogue', 'Department',
       'Classification', 'Object Number', 'Diameter (cm)',
       'Circumference (cm)', 'Height (cm)', 'Length (cm)', 'Width (cm)',
       'Depth (cm)', 'Weight (kg)', 'Duration (s)', 'Medium','Dimensions' ], inplace=True)

# Drop the rows that do not have the Acquisition Date or the Artist name

dfb4.dropna(subset=['Acquisition Date','Name'  ], inplace=True)

In [29]:
dfa4.columns

Index(['Artist ID', 'Name', 'Nationality', 'Gender', 'Birth Year',
       'Death Year'],
      dtype='object')

In [30]:
# Drop the columns we do not need for answering this question
dfa4.drop(columns = ['Nationality', 'Gender'])

# Drop the rows with a missing value on one of this
dfa4.drop(columns=['Gender', 'Nationality' ], inplace=True)

In [31]:
# As we are going to operate with dates, we need to have everything in the same format

In [32]:
# The column "Acquisition Date" is not a float as it is the 
dfb4["Acquisition Date"]

0         1996-04-09
1         1995-01-17
2         1997-01-15
3         1995-01-17
4         1997-01-15
             ...    
130254    2008-10-08
130256    2008-10-08
130257    2008-10-08
130259    2008-10-08
130261    2008-10-08
Name: Acquisition Date, Length: 123494, dtype: object

In [33]:
# We are going to extract the year of Acquisition from the Acquisition Date so we can see if it is in between the Birth Year and the Death Year
dfb4["Acquisition_Year"] = dfb4["Acquisition Date"].str.slice(0, 4)

In [34]:
# We convert from float to int the column BirthYear after eliminating the ones we do not have

dfa4.dropna(subset=['Birth Year'], inplace=True)
dfa4['Birth Year'] = dfa4['Birth Year'].astype(int)

In [35]:
# We fill the Death Year with 2024 and we will see from the result how can we proceed from there
# We also convert all values from float to int
dfa4['Death Year'] = dfa4['Death Year'].fillna(2024)
dfa4['Death Year'] = dfa4['Death Year'].astype(int)

In [36]:
dfb4.sort_values(by='Artist ID', ascending=True)

Artwork ID                                       Title Artist ID  \
32062       33599                         Study for Head Bath         1   
60218       64139                                General Nuke         1   
60378       64320                                  Moon Night        10   
64593       68977          Self Portrait - 55 Division Street       100   
75099       81066                         The Artist's Father       100   
...           ...                                         ...       ...   
3297         4214                                     Tumbler      9972   
1731         2489                  Dictating Unit (model 274)      9973   
2480         3346  Executary Transcribing Machine (model 272)      9973   
56935       60588                     Jane Reed and Dora Hunt       998   
41542       44196                 Tubercular Mother and Child       999   

                                                    Name              Date  \
32062                                     Robert Arneson              1977   
60218                                     Robert Arneson              1986   
60378                                      Irene Aronson              1951   
64593                          Ivan Le Lorraine Albright              1947   
75099                          Ivan Le Lorraine Albright  1935 (cast 1952)   
...                                                  ...               ...   
3297   Hemco Plastics Div. of Bryant Electrical Co., ...           c. 1939   
1731                          Eliot Noyes and Associates              1971   
2480                          Eliot Noyes and Associates              1966   
56935                                 Clarence H. Carter              1950   
41542                                        Paul Carter           c. 1937   

      Acquisition Date Acquisition_Year  
32062       1981-04-28             1981  
60218       1997-05-28             1997  
60378       1952-11-07             1952  
64593       1953-06-22             1953  
75099       1952-10-21             1952  
...                ...              ...  
3297        1939-12-01             1939  
1731        1973-05-08             1973  
2480        1969-04-24             1969  
56935       1953-01-30             1953  
41542       1941-01-16             1941  

[123494 rows x 7 columns]

In [37]:
dfa4.columns

Index(['Artist ID', 'Name', 'Birth Year', 'Death Year'], dtype='object')

In [38]:
dfb4.columns

Index(['Artwork ID', 'Title', 'Artist ID', 'Name', 'Date', 'Acquisition Date',
       'Acquisition_Year'],
      dtype='object')

In [39]:
dfa4['Artist ID'] = pd.to_numeric(dfa4['Artist ID'], errors='coerce')
dfb4['Artist ID'] = pd.to_numeric(dfb4['Artist ID'], errors='coerce')

In [40]:
filas_con_nan = dfb4[dfb4['Artist ID'].isnull()]

In [41]:
filas_con_nan.shape # HAY QUE COPIAR LAS 6K FILAS Y PONERLES EL SEGUNDO NÚMERO, COPIANDO EL RESTO DE INFO IGUAL

(6251, 7)

In [42]:
# Drop the null values from 
dfa4.dropna(subset=['Artist ID'], inplace=True)
dfb4.dropna(subset=['Artist ID'], inplace=True)
# We make sure that the Artist ID column from both dfs are the same type
dfa4['Artist ID'] = dfa4['Artist ID'].astype('int64')
dfb4['Artist ID'] = dfb4['Artist ID'].astype('int64')
# We merge both datasets by the column Artist ID
dfc4 = pd.merge(dfa4, dfb4, on='Artist ID', how='inner')

In [43]:
dfc4.drop(columns = ["Acquisition Date", 'Date'], inplace = True)

In [ ]:
dfc4["Acquisition_Year"] = dfc4["Acquisition_Year"].astype('int64')

In [44]:
dfc4

Artist ID           Name_x  Birth Year  Death Year  Artwork ID  \
0               1   Robert Arneson        1930        1992       33599   
1               1   Robert Arneson        1930        1992       64139   
2               2   Doroteo Arnaiz        1936        2024       61629   
3               3      Bill Arnold        1941        2024       45972   
4               3      Bill Arnold        1941        2024       45997   
...           ...              ...         ...         ...         ...   
110238      67352  Neelon Crawford        1946        2024      215740   
110239      67352  Neelon Crawford        1946        2024      215741   
110240      67352  Neelon Crawford        1946        2024      215742   
110241      67379          N. Dash        1980        2024      215840   
110242      67437  Sebastian Black        1985        2024      216351   

                                Title           Name_y Acquisition_Year  
0                 Study for Head Bath   Robert Arneson             1981  
1                        General Nuke   Robert Arneson             1997  
2                          BAS-RELIEF   Doroteo Arnaiz             1965  
3                    Honey under Sink      Bill Arnold             1972  
4                   Honey under Chair      Bill Arnold             1972  
...                               ...              ...              ...  
110238     Computer Graphics Lab Demo  Neelon Crawford             2016  
110239                    The Vincent  Neelon Crawford             2016  
110240  Ground Zero with Rick Schmidt  Neelon Crawford             2016  
110241                        Hot Eye          N. Dash             2016  
110242                          Tails  Sebastian Black             2016  

[110243 rows x 8 columns]

In [58]:
dfd4 = (dfc4['Acquisition_Year'] >= dfc4['Birth Year']) & (dfc4['Acquisition_Year'] <= dfc4['Death Year'])

In [59]:
dfd4 = pd.DataFrame(dfd4)

In [61]:
rows = []
for index, row in dfc4.iterrows():
    if row['Acquisition_Year'] >= row['Birth Year'] and row['Acquisition_Year'] <= row['Death Year']:
        rows.append(row)
dfd4 = pd.DataFrame(rows)
dfd4

Artist ID           Name_x  Birth Year  Death Year  Artwork ID  \
0               1   Robert Arneson        1930        1992       33599   
2               2   Doroteo Arnaiz        1936        2024       61629   
3               3      Bill Arnold        1941        2024       45972   
4               3      Bill Arnold        1941        2024       45997   
5               3      Bill Arnold        1941        2024       46023   
...           ...              ...         ...         ...         ...   
110238      67352  Neelon Crawford        1946        2024      215740   
110239      67352  Neelon Crawford        1946        2024      215741   
110240      67352  Neelon Crawford        1946        2024      215742   
110241      67379          N. Dash        1980        2024      215840   
110242      67437  Sebastian Black        1985        2024      216351   

                                Title           Name_y  Acquisition_Year  
0                 Study for Head Bath   Robert Arneson              1981  
2                          BAS-RELIEF   Doroteo Arnaiz              1965  
3                    Honey under Sink      Bill Arnold              1972  
4                   Honey under Chair      Bill Arnold              1972  
5                  Honey Walking Path      Bill Arnold              1972  
...                               ...              ...               ...  
110238     Computer Graphics Lab Demo  Neelon Crawford              2016  
110239                    The Vincent  Neelon Crawford              2016  
110240  Ground Zero with Rick Schmidt  Neelon Crawford              2016  
110241                        Hot Eye          N. Dash              2016  
110242                          Tails  Sebastian Black              2016  

[72663 rows x 8 columns]

In [64]:
df_artworks_count = dfd4.groupby('Name_y').size().reset_index(name='Number_of_artworks')
df_artworks_count.rename(columns={'Name_y': 'Name'}, inplace=True)
df_artworks_count

Name  Number_of_artworks
0             A. E. Gallatin                   1
1           A. F. Gangkofner                   2
2             A. G. Fronzoni                   8
3            A. M. Cassandre                  38
4            A. Michael Noll                   4
...                      ...                 ...
7752        Édouard Vuillard                   2
7753  Édouard-Wilfred Buquet                   1
7754              Éric Chahi                   1
7755           Étienne Hajdu                  21
7756        Öyvind Fahlström                   9

[7757 rows x 2 columns]